manifests/init.pp

```diff
@@ -110,7 +110,7 @@
   Optional[String] $archive_source,
   String $cfg_location,
   Hash $cfg,
-  Optional[Hash] $ldap_cfg,
+  Optional[Variant[Hash,Array]] $ldap_cfg,
```

Comment:
It would be great if we build a custom data type here.

Reply:
I would also be fine with:
```suggestion
  Optional[Variant[Hash[String[1],String[1]],Array][String[1]]] $ldap_cfg,
```

(assuming that the content is always strings)

Reply:
ping @unki :)